### PR TITLE
feat:add-prereq-dns-checks

### PR DESCRIPTION
**Pull request by andybeltz**
### Summary

- Added DNS entries pre-checks to the `Test-VrslcmPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-GlobalWsaPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-IlaPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-IomPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-PcaPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-PdrPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-HrmPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-CbrPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.
- Added DNS entries pre-checks to the `Test-CcmPrerequisite` cmdlet to ensure all assocated DNS entries are resolvable for both forward and reverse lookups.

### Type

- [ ] Bugfix
- [X] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [X] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### Test and Documentation

- [X] Tests have been completed.
- [ ] Documentation has been added or updated.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -13481,6 +13487,12 @@ Function Test-IlaPrerequisite {
                         Test-PrereqLicenseKey -licenseKey $jsonInput.licenseKey -productName "VMware Aria Suite or VMware Aria Operations for Logs" # Verify a license key is present
                         Test-PrereqServiceAccount -user ($jsonInput.domainBindUser + "@" + $jsonInput.domainFqdn) -password $jsonInput.domainBindPass -server ($jsonInput.domainControllerMachineName + "." + $jsonInput.domainFqdn) -domain $jsonInput.domainFqdn # Verify that the required service accounts are created in Active Directory
                         Test-PrereqDomainController -server ($jsonInput.domainControllerMachineName + "." + $jsonInput.domainFqdn) # Verify that Active Directory Domain Controllers are available in the environment
+                        # Verify that DNS Entries are resolvable in the environment
+                        $dnsEntries = $jsonInput.clusterFqdn, $jsonInput.hostNameNodeA, $jsonInput.hostNameNodeB, $jsonInput.hostNameNodeC
```
[the remark this answers is not in the document]
Comment (andybeltz) [reply PowerValidatedSolutions.psm1:13491]: I tested this - it doesnt appear to work in the context of the current setup, whereas the current code does.  I propose lets keep the current working code for now and if we need to make updates in the future we can do so.

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -2837,6 +2837,12 @@ Function Test-PdrPrerequisite {
                     Test-PrereqAriaAutomation # Verify that VMware Aria Automation has been deployed
                     Test-PrereqActiveDirectoryIntegration -server $jsonInput.protected.sddcManagerFqdn -user $jsonInput.protected.sddcManagerUser -password $jsonInput.protected.sddcManagerPass -domain $jsonInput.domainFqdn # Verify that VMware Cloud Foundation is integrated with Active Directory
                     Test-PrereqDomainController -server ($jsonInput.domainControllerMachineName + "." + $jsonInput.domainFqdn) # Verify that Active Directory Domain Controllers are available in the environment
+                    # Verify that DNS Entries are resolvable in the environment
+                    $dnsEntries = $jsonInput.protected.vrmsFqdn, $jsonInput.protected.srmFqdn, $jsonInput.recovery.vrmsFqdn, $jsonInput.recovery.srmFqdn
```
[the remark this answers is not in the document]
Comment (andybeltz) [reply PowerValidatedSolutions.psm1:2841]: I tested this - it doesnt appear to work in the context of the current setup, whereas the current code does.  I propose lets keep the current working code for now and if we need to make updates in the future we can do so.

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -17397,6 +17409,12 @@ Function Test-IomPrerequisite {
                         Test-PrereqBinary -searchCriteria "Operations-Manager-Appliance-$ariaOperationsVersion" -productMessage "VMware Aria Operations" # Verify that the required binaries are available
                         Test-PrereqLicenseKey -licenseKey $jsonInput.licenseKey -productName "VMware Aria Suite or VMware Aria Operations" # Verify a license key is present
                         Test-PrereqDomainController -server ($jsonInput.domainControllerMachineName + "." + $jsonInput.domainFqdn) # Verify that Active Directory Domain Controllers are available in the environment
+                        # Verify that DNS Entries are resolvable in the environment
+                        $dnsEntries = $jsonInput.clusterFqdn, $jsonInput.hostNameNodeA, $jsonInput.hostNameNodeB, $jsonInput.hostNameNodeC, $jsonInput.hostNameProxyA, $jsonInput.hostNameProxyB
```
[the remark this answers is not in the document]
Comment (andybeltz) [reply PowerValidatedSolutions.psm1:17413]: I tested this - it doesnt appear to work in the context of the current setup, whereas the current code does.  I propose lets keep the current working code for now and if we need to make updates in the future we can do so.

**Review by tenthirtyam — APPROVED** [review]
Cookie Monster

**Pull request merged**